### DISCUSSION TITLE
retrieve should allow for specifying fields to retrieve

### DISCUSSION THREAD

**Discussion by kgoerlitz** [jsforce/jsforce#1548] [Ideas]
The retrieve function should probably support a fields parameter that takes an array of field names (or a single field name).  If the fields are not specified then all of the fields should be retrieved.

**Comment by techieshark** [comment]
Still useful request.

A work around, with `findOne()`:

```
await connection
      .sobject("Contact")
      .findOne({ Id: idToRetrieve }, {
        Some_Field__c: true,
        Some_BuiltIn_Field: true,
      });
```

or similarly this probably works

```
await connection
      .sobject("Contact")
      .findOne( { Id: idToRetrieve }, ['Some_Field__c', 'Some_BuiltIn_Field'] });
```

**Comment by cristiand391** [comment]
closing this, v3 supports passing fields in the `options` object:
https://jsforce.github.io/jsforce/classes/sobject.SObject.html#retrieve
https://jsforce.github.io/jsforce/types/types_common.RetrieveOptions.html
